python/lsst/pipe/tasks/selectImages.py

```diff
@@ -590,3 +591,119 @@ def run(self, wcsList, bboxList, coordList, psfList, dataIds, **kwargs):
+    nImagesMax = pexConfig.RangeField(
+        dtype=int,
+        doc="Maximum number of images to select",
+        default=5,
+        min=0)
+    maxPsfFwhm = pexConfig.Field(
+        dtype=float,
+        doc="Maximum PSF FWHM (in arcseconds) to select",
+        default=1.5,
+        optional=True)
+    minPsfFwhm = pexConfig.Field(
+        dtype=float,
+        doc="Minimum PSF FWHM (in arcseconds) to select",
+        default=0.,
+        optional=True)
+
+
+class BestSeeingSelectVisitsTask(pipeBase.PipelineTask):
+    ConfigClass = BestSeeingSelectVisitsConfig
+    _DefaultName = 'bestSeeingSelectVisits'
+
+    def run(self, sourceTables):
+        inputVisits = [sourceTable.ref.dataId['visit'] for sourceTable in sourceTables]
+        psfSizes = []
+        visits = []
+        for visit, sourceTable in zip(inputVisits, sourceTables):
+            df = sourceTable.get(parameters={"columns": ['IxxPsf', 'IyyPsf', 'IxyPsf',
+                                                         'LocalWcs_CDMatrix_2_1', 'LocalWcs_CDMatrix_1_1',
+                                                         'LocalWcs_CDMatrix_1_2', 'LocalWcs_CDMatrix_2_2']})
+            wcsDet = (df.LocalWcs_CDMatrix_1_1 * df.LocalWcs_CDMatrix_2_2
+                      - df.LocalWcs_CDMatrix_1_2 * df.LocalWcs_CDMatrix_1_2)
+            # if min/max PSF values are defined, remove images out of bounds
+            pixToArcseconds = np.nanmedian(3600*np.degrees(np.sqrt(np.fabs(wcsDet))))
+            determinantRadius = np.power(np.nanmedian(df.IxxPsf * df.IxxPsf - df.IxyPsf**2), 0.25)
+            psfSize = determinantRadius*pixToArcseconds
+            sizeFwhm = psfSize * np.sqrt(8.*np.log(2.))
+            if self.config.maxPsfFwhm and sizeFwhm > self.config.maxPsfFwhm:
+                continue
+            if self.config.minPsfFwhm and sizeFwhm < self.config.minPsfFwhm:
+                continue
+            psfSizes.append(sizeFwhm)
+            visits.append(visit)
+
+        sortedVisits = [ind for (_, ind) in sorted(zip(psfSizes, visits))]
+        output = sortedVisits[:self.config.nImagesMax]
+        self.log.info(f"{len(output)} images selected with FWHM "
+                      f"range of {psfSizes[visits.index(output[0])]}"
+                      f"--{psfSizes[visits.index(output[-1])]} arcseconds")
+        # In order to store as a StructuredDataDict, convert list to dict
+        goodVisits = {key: True for key in output}
+        print(goodVisits)
+        return pipeBase.Struct(goodVisits=goodVisits)
+
+
+class BestSeeingPercentileSelectVisitsConfig(pipeBase.PipelineTaskConfig,
+                                             pipelineConnections=BestSeeingSelectVisitsConnections):
+    percentile = pexConfig.RangeField(
+        doc="Select top percentile of BestPercentile seeing visits. 1 selects all visits. 0 selects None",
+        dtype=float,
+        default=0.33,
+        min=0,
+        max=1,
+    )
+    nMin = pexConfig.Field(
```

Comment:
Please call this `nImagesMin` both for inherent clarity and to more naturally exist alongside `nImagesMax`. Incidentally... should these actually be `nVisitsMin` and `nVisitsMax`?